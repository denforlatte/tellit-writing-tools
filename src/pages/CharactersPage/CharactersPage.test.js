import React from 'react';
import { shallow } from 'enzyme';
import { CharactersPage } from './CharactersPage';

const characters = {
  isLoading: true,
};

const match = {
  params: {
    url: ''
  }
}

const mockFunction = jest.fn();

describe('Characters View', () => {
  describe('renders', () => {
    it('shows a spinner when fetching characters', () => {
      const wrapper = shallow(<CharactersPage characters={characters} match={match} getCharacters={mockFunction} selectCharacter={mockFunction} addNewCharacter={mockFunction} user={{}}/>);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
