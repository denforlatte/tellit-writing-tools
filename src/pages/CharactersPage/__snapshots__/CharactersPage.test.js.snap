// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters View renders shows a spinner when fetching characters 1`] = `
<Container
  tag="div"
>
  <h2>
    Major Characters
  </h2>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Spinner
      color="primary"
      size="lg"
      tag="div"
      type="border"
    >
      Loading...
    </Spinner>
  </div>
  <br />
  <h2>
    Minor Characters
  </h2>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Spinner
      color="primary"
      size="lg"
      tag="div"
      type="border"
    >
      Loading...
    </Spinner>
  </div>
  <br />
  <br />
  <br />
</Container>
`;
